src/Common/src/System/Net/CaseInsensitiveAscii.cs

```diff
@@ -43,7 +43,7 @@ internal class CaseInsensitiveAscii : IEqualityComparer, IComparer
         public int GetHashCode(object myObject)
         {
             string myString = myObject as string;
-            if (myObject == null)
+            if (myString == null)
```

Comment:
Roslyn is smart enough to avoid the string equality operator when comparing null, so object.ReferenceEquals isn't necessary.